TransformContinuousIT.testContinousEvents test failed

**Build scan**:
https://gradle-enterprise.elastic.co/s/rpnsd3e5jwf3y

**Repro line**:
```
REPRODUCE WITH: ./gradlew ':x-pack:plugin:transform:qa:single-node-tests:javaRestTest' --tests "org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.testContinousEvents" \
  -Dtests.seed=DA001B314F2CA42D \
  -Dtests.security.manager=true \
  -Dtests.locale=ar-DZ \
  -Dtests.timezone=Asia/Jayapura \
  -Druntime.java=8

REPRODUCE WITH: ./gradlew ':x-pack:plugin:transform:qa:single-node-tests:javaRestTest' --tests "org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.testContinousEvents" \
  -Dtests.seed=DA001B314F2CA42D \
  -Dtests.security.manager=true \
  -Dtests.locale=ar-DZ \
  -Dtests.timezone=Asia/Jayapura \
  -Druntime.java=8
```

**Reproduces locally?**:
No

**Applicable branches**:
7.x

**Failure history**:
This is the first failure since August 6th.

**Failure excerpt**:
```
13:49:46 org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT > testContinousEvents FAILED
13:49:46     java.lang.AssertionError: Error in test case [continuous-terms-pivot-test].If you want to mute the test, please mute [org.elasticsearch.xpack.transform.integration.continuous.TermsGroupByIT] only, but _not_ [org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT] as a whole.
13:49:46         at __randomizedtesting.SeedInfo.seed([DA001B314F2CA42D:E1366A2AC547267B]:0)
13:49:46         at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.testContinousEvents(TransformContinuousIT.java:258)
13:49:46 
13:49:46         Caused by:
13:49:46         java.lang.AssertionError: Ingest run: 1 did not match max run: 0.0, iteration: 1
13:49:46         Expected: <0.0>
13:49:46              but: was <1.0>
13:49:46             at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
13:49:46             at org.junit.Assert.assertThat(Assert.java:956)
13:49:46             at org.elasticsearch.xpack.transform.integration.continuous.TermsGroupByIT.testIteration(TermsGroupByIT.java:144)
13:49:46             at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.testContinousEvents(TransformContinuousIT.java:249)
```

--- a/x-pack/plugin/transform/qa/single-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/transform/integration/continuous/TransformContinuousIT.java
+++ b/x-pack/plugin/transform/qa/single-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/transform/integration/continuous/TransformContinuousIT.java
@@ -237,7 +237,10 @@ public class TransformContinuousIT extends ESRestTestCase {
 
             // start all transforms, wait until the processed all data and stop them
             startTransforms();
-            waitUntilTransformsReachedUpperBound(runDate.getEpochSecond() * 1000 + 1);
+
+            // at random we added between 0 and 999_999ns == (1ms - 1ns) to every data point, so we add 1ms, so every data point is before
+            // the checkpoint
+            waitUntilTransformsReachedUpperBound(runDate.toEpochMilli() + 1, run);
             stopTransforms();
 
             // TODO: the transform dest index requires a refresh, see gh#51154
@@ -387,11 +390,12 @@ public class TransformContinuousIT extends ESRestTestCase {
         }
     }
 
-    private void waitUntilTransformsReachedUpperBound(long timeStampUpperBoundMillis) throws Exception {
+    private void waitUntilTransformsReachedUpperBound(long timeStampUpperBoundMillis, int iteration) throws Exception {
         logger.info(
-            "wait until transform reaches timestamp_millis: {}",
+            "wait until transform reaches timestamp_millis: {} iteration: {}",
             ContinuousTestCase.STRICT_DATE_OPTIONAL_TIME_PRINTER_NANOS.withZone(ZoneId.of("UTC"))
-                .format(Instant.ofEpochMilli(timeStampUpperBoundMillis))
+                .format(Instant.ofEpochMilli(timeStampUpperBoundMillis)),
+            iteration
         );
         for (ContinuousTestCase testCase : transformTestCases) {
             assertBusy(() -> {

